tests/address_space/address_space_common.h

```diff
@@ -0,0 +1,198 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provide common code for address_space verification
+//
+*******************************************************************************/
+
+#ifndef SYCL_CTS_TESTS_ADDRESS_SPACE_ADDRESS_SPACE_COMMON_H
+#define SYCL_CTS_TESTS_ADDRESS_SPACE_ADDRESS_SPACE_COMMON_H
+
+#include "../common/common.h"
+
+#include <array>
+#include <string>
+#include <type_traits>
+
+#define EXPECT_EQUALS(lhs, rhs) \
+  if ((lhs) != (rhs)) return false;
+
+#define EXPECT_ADDRSPACE_EQUALS(expr, as) \
+  EXPECT_EQUALS(expr, (AddrSpace<as>()).get_value())
+
+#ifndef TEST_NAME
+#error Invalid test namespace
+#endif
+
+namespace TEST_NAMESPACE {
+using namespace sycl_cts;
+
+template <typename T>
+struct address_space_core_kernel {};
+
+template <typename T> class check_types {
+public:
+  template <cl::sycl::access::address_space kAs>
+  struct AddrSpace {
+    constexpr T get_value() const { return static_cast<T>(to_integral(kAs)); }
+  };
+
+  // the parameter has no address space specification, so it'll
+  // get duplicated based on the argument's address space
+  // it should call the appropiate overload of f(), and that will
+  // tell us the address space the parameter was in
+
+  static T readValue(T *p) { return *p; }
+
+  static T readValue(T &p) { return p; }
+
+  struct SomeStruct {
+    T readValue(T *p) { return *p; }
+
+    T readValue(T &p) { return p; }
+
+    T operator[](T *p) { return *p; }
+
+    T operator[](T &p) { return p; }
+  };
+
+  template <cl::sycl::access::address_space AspSpace>
+  static bool test_duplication(cl::sycl::multi_ptr<T, AspSpace> ptr) {
+    EXPECT_ADDRSPACE_EQUALS(readValue(ptr), AspSpace);
+    EXPECT_ADDRSPACE_EQUALS(readValue(*ptr), AspSpace);
+
+    SomeStruct d;
+    EXPECT_ADDRSPACE_EQUALS(d.readValue(ptr), AspSpace);
+    EXPECT_ADDRSPACE_EQUALS(d.readValue(*ptr), AspSpace);
+    EXPECT_ADDRSPACE_EQUALS(d[ptr], AspSpace);
+    EXPECT_ADDRSPACE_EQUALS(d[*ptr], AspSpace);
+
+    return true;
+  }
+
+  static bool test_duplication(cl::sycl::global_ptr<T> globalPtr,
+                        cl::sycl::local_ptr<T> localPtr,
```

Comment:
Indent.
It looks like most of the function declaration have weird indent.

Reply:
Sorry, applied clang-format in b1aec96c504d3f4de726ba75404ba498e418b8ef